File: spring-web/src/main/java/com/example/springwebex/controller/DvdRentalController.java
package com.example.springwebex.controller;

import java.util.List;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.springwebex.model.rdb.ActorDto;
import com.example.springwebex.service.ActorService;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@RestController
@RequiredArgsConstructor
@RequestMapping("/dvdrental")
@Slf4j
public class DvdRentalController {
    private final ActorService actorService;

    @GetMapping("/actors")
    public List<ActorDto> getActors() {
        var result = actorService.getActors();
        log.info("{}", result);
        return result;
    }

    @GetMapping("/actor/{id}")
    public ActorDto getActorById(@PathVariable("id") Long id) {
        var result = actorService.getActorById(id);
        log.info("getActors{}", result);
        return result;
    }

    @GetMapping("/actor/insert/{firstName}/{lastName}")
    public void setActor(@PathVariable("firstName") String firstName,
            @PathVariable("lastName") String lastName) {
        actorService.setActor(firstName, lastName);
        log.info("insertActor()");
    }
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/controller/DvdRentalController.kt
package com.example.springwebex.controller

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import com.example.springwebex.model.rdb.ActorDto
import com.example.springwebex.service.ActorService

@RestController
@RequestMapping("/dvdrental")
class DvdRentalController(private val actorService: ActorService) {

    private val log = KotlinLogging.logger {}

    @GetMapping("/actors")
    fun getActors(): List<ActorDto> {
        val result = actorService.getActors()
        log.info { "$result" }
        return result
    }

    @GetMapping("/actor/{id}")
    fun getActorById(@PathVariable("id") id: Long): ActorDto {
        val result = actorService.getActorById(id)
        log.info { "getActors$result" }
        return result
    }

    @GetMapping("/actor/insert/{firstName}/{lastName}")
    fun setActor(
        @PathVariable("firstName") firstName: String,
        @PathVariable("lastName") lastName: String
    ) {
        actorService.setActor(firstName, lastName)
        log.info { "insertActor()" }
    }
}
